File: Generating/singleton/python/Person.py
class Person():
  __instance = None

  @staticmethod
  def getInstance():
    if Person.__instance == None:
      Person.__instance = Person()
    
    return Person.__instance

  def __init__(self):
    self.__human = {
      'name': 'Ivan',
      'age': 23
    }
    pass

  def getHuman(self):
    return self.__human
  
  def setHuman(self, name, age):
    self.__human['name'] = name
    self.__human['age'] = age
    pass

File: Generating/singleton/src/Person.ts
import { IPerson, Human } from './interfaces'

let person = null

class Person implements IPerson {
  private human: Human
  constructor() {
    this.human = { name: 'Ivan', age: 23 }
  }

  setHuman(
    newName: string,
    newAge: number
  ): void {
    if (newName && newAge) {
      this.human.name = newName
      this.human.age = newAge
    }
  }

  getHuman(): Human {
    return this.human
  }
}

export default function newPerson() {
  if (person === null) {
    person = new Person
  }

  return person
}
